pkg/datapath/linux/probes/probes.go

```diff
@@ -205,6 +205,42 @@ func (p *ProbeManager) SystemConfigProbes() error {
 	return nil
 }
 
+// GetRequiredConfig performs a check of mandatory kernel configuration options. It
+// returns a map indicating which required kernel paramaters are enabled - and which are not
+// GetRequiredConfig is being used by CLI "cilium kernel-check"
+func (p *ProbeManager) GetRequiredConfig() (supported map[KernelParam]bool) {
+	config := p.features.SystemConfig
+
+	if config.ConfigBpf == "" {
+		fmt.Println("Unable to detect BPF support. Is the kernel config available?\n")
+	}
```

Comment:
RIght, I had missed the check on my previous review. The check itself looks good. Please remove the trailing `\n` from the message, the `Println()` should already take care of the new line.

I'm not sure about the location of the check, though. There's no reason why we would have it in `GetRequiredConfig()` and not in `GetOptionalConfig()`. We could have it in both, maybe, and it should probably return an error instead of trying to fill the `kernelParams` map if config file is missing.

Another possibility would be to move it to a separate function, to call that function from `kernelCheck()` and to exit early if we think we don't have configuration info.

Reply:
@qmonnet decided to move the check to a seperate function and exit early. what do you think of this approach?